torchmetrics/metric.py

```diff
@@ -481,107 +481,108 @@ def __hash__(self):
 
         return hash(tuple(hash_vals))
 
-    def __add__(self, other: Any):
+    def __add__(self, other: "Metric") -> "Metric":
         return CompositionalMetric(torch.add, self, other)
 
-    def __and__(self, other: Any):
+    def __and__(self, other: "Metric") -> "Metric":
         return CompositionalMetric(torch.bitwise_and, self, other)
 
-    def __eq__(self, other: Any):
+    # Fixme: this shall return bool instead of Metric
```

Comment:
@justusschock any idea how we could fix this?

Reply:
I discussed with @tchaton  in the past. One can argue for bool and for metric (the metric checks the equality of the outputs for each argument then). Equality would probably go via `is` or hashes.

So the current behaviour at least is consistent with the other arithmetic and logic operators.

Reply:
yep, I see the arithmetic but feel that logic like eq shall return a bool, that is also why mypy complains as we overwrite something which shall return bool but return metric...

Reply:
How strict are we with deprecations in metrics? I'm fine with changing it. But I think checking the equality of metrics is not that easy. Probably it would require some syncing and checking the states.

Reply:
we keep 0.1 deprecation loop, but if needed we are fine with breaking changes